source/extract.cpp

```diff
@@ -24,72 +26,67 @@
 namespace i18n = brls::i18n;
 using namespace i18n::literals;
 
+constexpr size_t WRITE_BUFFER_SIZE = 0x1000000;
+
 namespace extract {
 
     namespace {
         bool caselessCompare(const std::string& a, const std::string& b)
         {
             return strcasecmp(a.c_str(), b.c_str()) == 0;
         }
-
-        void preWork(zipper::Unzipper& unzipper, const std::string& workingPath, std::vector<zipper::ZipEntry>& entries)
-        {
-            chdir(workingPath.c_str());
-            entries = unzipper.entries();
-            s64 uncompressedSize = 0;
-            s64 freeStorage;
-            for (const auto& entry : entries)
-                uncompressedSize += entry.uncompressedSize;
-
-            if (R_SUCCEEDED(fs::getFreeStorageSD(freeStorage))) {
-                if (uncompressedSize * 1.1 > freeStorage) {
-                    unzipper.close();
-                    brls::Application::crash("menus/errors/insufficient_storage"_i18n);
-                    std::this_thread::sleep_for(std::chrono::microseconds(2000000));
-                    brls::Application::quit();
-                }
-            }
```

Comment:
this is a feature, extracting a file on a full sd would lead to a bunch of complications.
You should be able to get the uncompressed size from the headers if minizip doesn't offer it natively https://en.wikipedia.org/wiki/ZIP_(file_format)#File_headers
I already implemented some code to retrieve data from headers so you could take inspiration from it https://github.com/HamletDuFromage/aio-switch-updater/blob/master/source/utils.cpp#L24

Reply:
Added with unzipper. I lazy to read again the minizip doc